app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -30,11 +31,9 @@ def call
 
       private
 
-      def validate_xms_mirid_format
+      def parse_xms_mirid
         begin
-          required_keys = %w(subscriptions resourcegroups providers)
-          missing_keys = required_keys - xms_mirid_hash.keys
-          raise Err::ClaimInInvalidFormat, "Required keys #{missing_keys} are missing" if missing_keys.empty?
+          xms_mirid_hash
```

Comment:
In this function(xms_mirid_hash)  you dont raise exception . You prefer to handle it separately on the caller function? The problem is we will not know that the problem was with Hash creating.